api/pkg/artifact/artifact_test.go

```diff
@@ -49,19 +71,48 @@ func TestGcsArtifactClient_ParseURL(t *testing.T) {
 			},
 			wantErr: false,
 		},
+		{
+			name:      "invalid url",
+			urlScheme: "gs",
+			args: args{
+				gsURL: "!@#$%^&*()",
+			},
+			wantErr: true,
+		},
+		{
+			name:      "invalid scheme",
+			urlScheme: "gs",
+			args: args{
+				gsURL: "s3://bucket-name/object-path/object-path-2/object-path-3/file-1.txt",
+			},
+			wantErr: true,
+		},
+		{
+			name:      "empty bucket",
+			urlScheme: "gs",
+			args: args{
+				gsURL: "gs:///object-path/object-path-2/object-path-3/file-1.txt",
+			},
+			wantErr: true,
+		},
+		{
+			name:      "empty object",
+			urlScheme: "gs",
+			args: args{
+				gsURL: "gs://bucket-name/",
+			},
+			wantErr: true,
+		},
```

Comment:
These tests don't exactly correspond to any of the changes I made, but I added them since they're pretty to implement anyway (and we lacked them, so these tests also bump up the test coverage).